lock-api/src/main/java/com/palantir/lock/client/RequestBatchers.java

```diff
@@ -0,0 +1,76 @@
+import com.palantir.lock.watch.LockWatchEventCache;
+import com.palantir.lock.watch.NoOpLockWatchEventCache;
+import com.palantir.lock.watch.StartTransactionsLockWatchEventCache;
+import java.util.Optional;
+
+public final class RequestBatchers {
+    private final LockWatchEventCache lockWatchEventCache;
+    private final StartTransactionsLockWatchEventCache startTransactionsLockWatchEventCache;
+    private final Optional<Namespace> namespace;
+    private final Optional<MultiClientTransactionStarter> maybeTransactionStarter;
+    private final Optional<MultiClientCommitTimestampGetter> maybeCommitTimestampGetter;
+
+    private RequestBatchers(
+            LockWatchEventCache lockWatchEventCache,
+            Optional<Namespace> namespace,
+            Optional<MultiClientTransactionStarter> maybeTransactionStarter,
+            Optional<MultiClientCommitTimestampGetter> maybeCommitTimestampGetter) {
+        this.lockWatchEventCache = lockWatchEventCache;
+        this.startTransactionsLockWatchEventCache = StartTransactionsLockWatchEventCache.create(lockWatchEventCache);
+        this.namespace = namespace;
+        this.maybeTransactionStarter = maybeTransactionStarter;
+        this.maybeCommitTimestampGetter = maybeCommitTimestampGetter;
+    }
+
+    public static RequestBatchers create(
+            LockWatchEventCache lockWatchEventCache,
+            Optional<Namespace> namespace,
+            Optional<MultiClientTransactionStarter> maybeBatcher,
+            Optional<MultiClientCommitTimestampGetter> maybeCommitTimestampGetter) {
+        return new RequestBatchers(lockWatchEventCache, namespace, maybeBatcher, maybeCommitTimestampGetter);
+    }
+
+    public static RequestBatchers createForTests() {
+        return new RequestBatchers(
+                NoOpLockWatchEventCache.create(), Optional.empty(), Optional.empty(), Optional.empty());
+    }
+
+    public IdentifiedAtlasDbTransactionStarter getBatchingTransactionStarter(LockLeaseService lockLeaseService) {
+        if (!namespace.isPresent() || !maybeTransactionStarter.isPresent()) {
+            return BatchingIdentifiedAtlasDbTransactionStarter.create(
+                    lockLeaseService, startTransactionsLockWatchEventCache);
+        }
+        return new NamespacedIdentifiedTransactionStarter(
+                namespace.get(),
+                maybeTransactionStarter.get(),
+                startTransactionsLockWatchEventCache,
+                new LockCleanupService(lockLeaseService));
+    }
+
+    public CommitTimestampGetter getBatchingCommitTimestampGetter(LockLeaseService lockLeaseService) {
+        if (!namespace.isPresent() || !maybeCommitTimestampGetter.isPresent()) {
+            return BatchingCommitTimestampGetter.create(lockLeaseService, lockWatchEventCache);
+        }
+        return new NamespacedCommitTimestampGetter(
+                lockWatchEventCache, namespace.get(), maybeCommitTimestampGetter.get());
+    }
```

Comment:
Hmm, I think we need to clarify this a bit. Either

1. name these methods `create` rather than `get`, or
2. change the methods so that once an instance has been created for a given lock-lease service it always returns the same one.

`get` suggests that this is idempotent, but it's not: if you call `getBatchingTransactionStarter` twice with the same lock-lease service, you will get _two_ autobatchers!